### ISSUE TITLE
Asynchronous consensus proposals

### ISSUE THREAD

**Issue by elsirion** [fedimint/fedimint#4109]
Currently, if one module is slow generating its consensus proposal it block everything. With HBBFT this design made sense since it was progressing in epochs, with AlephBFT this isn't the case anymore and it might make sense to just propose module CIs whenever the module is ready and produces them.

Thoughts @joschisan?

Would fix #4101.

**Comment by joschisan** [comment]
Right now the only latency critical proposed item is the submission of decryption shares by the ln module. This will be removed in the ng ln module. Then, all CIs left are votes on feerate, blocktime and time, hence information that is used to validate transactions in the broadcast and therefore have to be on the broadcast. Then we can then only call consensus_proposal at the granularity we want to support, e.g. every five seconds and the module simply return all currently available items. Furthermore, using the broadcast for something else but information needed to validate incoming transaction, like signature shares for ecash or client config, might be tempting, but was a very unfortunate design decision every time we did it and led to unnecessary complexity inside and outside of the modules, increased latency and horrible inefficiencies. The current mechanism has been chosen to work in unison and encourage design patterns that make sense for our modules, though it might not seem ideal on first glance. Therefore, if there is a module for which this mechanism constitutes a problem in the future this could imply a suboptimal design on the modules part. I could add this reasoning to the documentation, since I have made this argument multiple times now it seems to be a common point of irritation. If then, we have an external module that actually requires a different mechanism to what we currently have I am happy to adapt it then.
